packages/ens-controller/src/EnsController.ts

```diff
@@ -194,6 +195,7 @@ export class EnsController extends BaseController<
   delete(chainId: Hex, ensName: string): boolean {
     const normalizedEnsName = normalizeEnsName(ensName);
     if (
+      !isSafeDynamicKey(chainId) ||
```

Comment:
Should we add a test for this? Seems important to check.

Reply:
It's true that `isValidHexAddress` for `address` and `isSafeChainId` for `chainId` would be sufficient validation to fix the violations currently found by the scanner. 

However, it seems worth it to implement more generalized blocklisting logic for this vulnerability (https://portswigger.net/web-security/prototype-pollution/preventing#sanitizing-property-keys), which will also directly placate the CodeQL scanner.

Reply:
Sure, we could definitely make more improvements that would ensure we can protect against this vulnerability more generally.

That seems unrelated to my comment, though. Should we add tests which attempt to call `delete` with `__proto__`, `constructor`, and `prototype` and verify that they don't do anything?

Reply:
I added tests for all three files here: https://github.com/MetaMask/core/pull/4041/commits/a145237447a84bf3fa534ad0bc54cd7b22601a37. Coverage is up to 100 for the three modules. I didn't test every string in the blocklist, but it seems like verifying that the validation works should be enough?

Reply:
Testing that it essentially just gets called is fine. Do we need tests for the function itself, though?

Reply:
Ah of course I'll add that right now.

Reply:
Tests added! https://github.com/MetaMask/core/pull/4041/commits/03333afd8517922f14cf0da4dc42b06f45e1991c